### PR TITLE
Implemented github actions: Alex

### LINKED ISSUE #1: Create Github Actions - Pytest
As a developer, I want to create a GitHub Actions testing workflow for the Starwatch project, so that automated tests are run on every code commit to ensure data pipeline stability.

- Create a githubs action pytest file

### LINKED ISSUE #2: Create Github actions - Pylint
As a developer, when we push new features or updated code to the repository, we want a suite of automated tests to run to check the code quality using pylint as part of a CI/CD pipeline.

### PR DESCRIPTION
**Pull request by AlexOBarnes**
Implemented github actions:
- Created pytest yml file
- Created pylint yml file

This closes #1  
This closes #2

### CODE REVIEW

**Comment by AlexOBarnes** [comment]
This setups up CI/CD tests for our project

File: .github/workflows/py_lint.yml [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+        uses: actions/setup-python@v1
+        with:
+          python-version: '3.12'
+
+      - name: "Find and install all requirements.txt files"
+        run: |
+          find . -name 'requirements.txt' -exec pip install -r {} \;
+
+      - name: Run pylint
+        run: pylint **/*.py --fail-under=8
```
Comment (olliejth) [inline .github/workflows/py_lint.yml:34]: Good idea to do it folder by folder.

File: .github/workflows/py_lint.yml [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+jobs:
+  lint:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Print Message
+        run: echo "New activity in repository detected"
+      
+      - name: checkout
+        uses: actions/checkout@v4
```
Comment (CalebMathers) [inline .github/workflows/py_lint.yml:22]: V4 very good, V3 doesn't work anymore!

File: .github/workflows/py_test.yml [hunk 3]
```diff
@@ -0,0 +1,28 @@
+on:
```
Comment (CalebMathers) [inline .github/workflows/py_test.yml:1]:
```suggestion
name: Test Code

on:
```

To keep it in form with the lint yaml

File: .github/workflows/py_lint.yml [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+    types: [opened, synchronize, reopened]
+
+
+jobs:
+  lint:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Print Message
+        run: echo "New activity in repository detected"
```
Comment (MurasatoNaoya) [inline .github/workflows/py_lint.yml:19]: Nice print statements to explain how the linting workflow is running.

File: .github/workflows/py_test.yml [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+    steps:
+    - name: "Checkout the repository"
+      uses: actions/checkout@v4
+
+    - name: "Install Python"
+      uses: actions/setup-python@v5.1.1
+      with:
+        python-version: '3.12'
+
+    - name: "Find and install all requirements.txt files"
```
Comment (MurasatoNaoya) [inline .github/workflows/py_test.yml:23]: All steps in the world flow clearly named to explicitly describe each stage.

File: dashboard/test_dashboard.py
Comment (CalebMathers) [inline dashboard/test_dashboard.py:1]: Linting our test files, fearless.

File: .github/workflows/py_test.yml [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+      run: |
+        find . -name 'requirements.txt' -exec pip install -r {} \;
```
Comment (CalebMathers) [inline .github/workflows/py_test.yml:25]: Good way to find all the requirements

File: .github/workflows/py_lint.yml (hunk 1 above)
Comment (CalebMathers) [inline .github/workflows/py_lint.yml:34]: 8 is a good cut off

**Review by MurasatoNaoya — APPROVED** [review]
Nice that this the the first pull request made, ensuring all later commits have a high linting score and (although the tests are basic now) creates a framework for automated CI/CD testing also.

**Review by CalebMathers — CHANGES_REQUESTED** [review]

File: .github/workflows/py_lint.yml (hunk 1 above)
Comment (AlexOBarnes) [reply .github/workflows/py_lint.yml:34]: Thanks caleb

**Review by olliejth — APPROVED** [review]
Well set-out starting structure for the repo as a whole, will help ensure quality as different features are added.

**Review by CalebMathers — APPROVED** [review]

**Pull request merged**